docs/book/src/developer/e2e.md

```diff
@@ -0,0 +1,228 @@
+- Manifests & docker images for the Cluster API infrastructure provider; in most cases
+  also machine images are required (AMI, OVA etc.)
+- Credentials for the target infrastructure provider 
+- Other support tools (e.g. kustomize, gsutil etc.)
+
+The Cluster API test framework provides support for building and retrieving the manifest
+files for Cluster API core components and for the Cluster API infrastructure provider
+(see [Setup](#setup))
+
+For the remaining tasks you can find examples of
+how this can be implemented e.g. in [CAPA E2E tests] and [CAPG E2E tests]. 
+
+## Setup
+
+In order to run E2E tests it is required to create a Kubernetes cluster with a 
+complete set of Cluster API providers installed. Setting up those elements is 
+usually implemented in a `BeforeSuite` function, and it consists of two steps:
+
+- Defining an E2E config file
+- Creating the management cluster and installing providers
+
+### Defining an E2E config file
+
+The [E2E config file] provides a convenient and flexible way to define common tasks for
+setting up a management cluster.
+
+Using the config file it is possible to:
+
+- Define the list of providers to be installed in the management cluster. Most notably, 
+  for each provider it is possible to define:
+  - One or more versions of the providers manifest (built from the sources, or pulled from a
+    remote location).
+  - A list of additional files to be added to the provider repository, to be used e.g.
+    to provide `cluster-templates.yaml` files.
+- Define the list of variables to be used when doing `clusterctl init` or
+  `clusterctl config cluster`.
+- Define a list of intervals to be used in the test specs for defining timeouts for the
+  wait commands/for the `Eventually` method.
+- Define the list of images to be loaded in the management cluster (this is specif of
+  management cluster based on kind).
+
+An [example E2E config file] can be found here.
+
+<aside class="note">
+
+<h1>Deprecated E2E config file format</h1>
+
+The [Cluster API test framework] includes also a [deprecated E2E config file] implementation,
+that was used before the introduction of clusterctl. This might be removed in future releases 
+of the test framework.
+
+</aside>
+
+### Creating the management cluster and installing providers
+
+In order to run Cluster API E2E tests, you need a Kubernetes cluster; the [NewKindClusterProvider] gives you a 
+type that can be used to create a local kind cluster and pre-load images into it, but also existing clusters can
+be used if available.
+
+Once you have a Kubernetes cluster, the [InitManagementClusterAndWatchControllerLogs method] provides a convenient
+way for installing providers.
+
+This method:
+- Runs `clusterctl init` using the above local repository.
+- Waits for the providers controllers to be running.
+- Creates log watchers for all the providers
+
+<aside class="note">
+
+<h1>Deprecated InitManagementCluster method</h1>
```

Comment:
Same comment as above. Instead of `<h1>`, `####` may be more suitable here.

Reply:
same